core/src/main/java/bio/terra/pearl/core/dao/participant/PortalParticipantUserDao.java

```diff
@@ -69,6 +69,9 @@ public List<PortalParticipantUser> findByProfileId(UUID profileId) {
         return findAllByProperty("profile_id", profileId);
     }
 
+    public Optional<PortalParticipantUser> findByParticipantUserIdAndProfileId(UUID userId, UUID profileId) {
```

Comment:
do we need to add the userId to this query?  I think it's redundant, especially since we've added the profileId uniqueness constraint to ppUser

Reply:
Ah, good catch.
Took me a while to get that 'profile_id' is the join key between these tables. Sorry about that.

Reply:
heh, as you can tell from the lack of previously existing constraints, helper methods, etc.., it also took me until today to realize that profile_id is functioning as the join key